model/registry/profile.yaml

```diff
@@ -0,0 +1,63 @@
+groups:
+
+  - id: registry.profile.frame
+    type: attribute_group
+    brief: >
+      Describes the origin of a single frame in a Profile.
+    attributes:
+      - id: profile.frame.type
+        stability: experimental
+        brief: >
+          Describes the interpreter or compiler of a single frame.
+        examples: ['cpython']
+        type:
+          allow_custom_values: true
+          members:
+            - id: dotnet
+              brief: >
+                [.NET](https://wikipedia.org/wiki/.NET)
+              value: "dotnet"
+              stability: experimental
+            - id: jvm
+              brief: >
+                [JVM](https://wikipedia.org/wiki/Java_virtual_machine)
+              value: "jvm"
+              stability: experimental
+            - id: kernel
+              brief: >
+                [Kernel](https://wikipedia.org/wiki/Kernel_(operating_system))
+              value: "kernel"
+              stability: experimental
+            - id: native
+              brief: >
+                [C](https://wikipedia.org/wiki/C_(programming_language)),
+                [C++](https://wikipedia.org/wiki/C%2B%2B),
+                [Go](https://wikipedia.org/wiki/Go_(programming_language)),
+                [Rust](https://wikipedia.org/wiki/Rust_(programming_language))
```

Comment:
Are these 4 languages grouped together on purpose, or because it's difficult for the ebpf profiler to distinguish them? Did we write down any use cases for these frame types somewhere? IIRC from the SIG discussions, the use case is to visually highlight different languages when visualizing the data (e.g. with different colors in a flame graph). If that's the case, distinguishing between C, C++, Go and Rust would probably be useful?

Apologies if this has been discussed already.

Reply:
These four named languages are just an example for `native` frames and the list is not exhaustive.
From a pure stack unwinding point of view, I think it might be hardly possible to distinguish them. But by examining the backing executable/shared library, it should be possible. At the moment, this additional inspection is not done and needs more research.

Overall, `native` is just a general grouping and I think there should be no issue being more specific in the future and adding more specific languages, like `rust`.

Reply:
Sounds good. Thanks!